### PR TITLE
[FIX] website_sale: Prevent display categories from other websites

### PR DESCRIPTION
**Pull request by abz-odoo**
Steps:
    - Go to website/configuration/ecommerce categories
    - Change `Chairs` website to `Website`
    - Go to `Website 2`/shop
    - Write `ch` in the search bar

Autocomplete returns `Chairs` even if it's not available in `Website 2`

opw-2884856

### CODE REVIEW

**Comment by bso-odoo** [comment]
It seems insufficient:
image
It works for removing the category entry itself, but not for the category link inside the products.

In `_search_render_results` of `product.template` you will need to filter the categories with the same domain in;
```py
            if with_category and product.public_categ_ids:
                data['category'] = {'extra_link_title': _('Categories:') if len(product.public_categ_ids) > 1 else _('Category:')}
```


For the commit message, instead of "Doesn't display", I would put "prevent display" (so it fits the sentence "this commit will ...").

File: addons/website_sale/models/product_template.py [hunk 1]
```diff
@@ -399,7 +399,9 @@ def _search_render_results(self, fetch_fields, mapping, icon, limit):
         with_category = 'extra_link' in mapping
         with_price = 'detail' in mapping
         results_data = super()._search_render_results(fetch_fields, mapping, icon, limit)
+        current_website = self.env['website'].get_current_website()
         for product, data in zip(self, results_data):
+            categ_ids = product.public_categ_ids.filtered(lambda m: not m.website_id or m.website_id == current_website)
```
Comment (bso-odoo) [inline addons/website_sale/models/product_template.py]: Maybe `c` for categories instead of `m` ?

**Review by bso-odoo — APPROVED** [review]
LGTM :+1: 
Maybe in message: either "prevent display of" or "prevent displaying"

File: addons/website_sale/models/product_template.py [hunk 2]
```diff
@@ -408,10 +410,10 @@ def _search_render_results(self, fetch_fields, mapping, icon, limit):
                     data['list_price'] = self.env['ir.qweb.field.monetary'].value_to_html(combination_info['list_price'], monetary_options)
             if with_image:
                 data['image_url'] = '/web/image/product.template/%s/image_128' % data['id']
-            if with_category and product.public_categ_ids:
+            if with_category and categ_ids:
                 data['category'] = {'extra_link_title': _('Categories:') if len(product.public_categ_ids) > 1 else _('Category:')}
```
Comment (Williambraecky) [inline addons/website_sale/models/product_template.py:414]:
```suggestion
                data['category'] = {'extra_link_title': _('Categories:') if len(categ_ids) > 1 else _('Category:')}
```

**Review by Williambraecky — CHANGES_REQUESTED** [review]
Last detail and then it's GTG 😄

**Comment by Williambraecky** [comment]
Thanks for your work!

robodoo r+

**Pull request closed without merging**